test/functorch/test_aotdispatch.py

```diff
@@ -694,7 +694,7 @@ def f(a, b, c, d):
             #
             # TODO(whc)- are the saved-tensors/saved-symints correct here?
             # i just made the test pass based on what default partition did
-            [False, True, True, False, False] + [False] * 5 + [True] * 3,
+            [False, True, True, False, False] + [False] * 4 + [True] * 4,
```

Comment:
Hey @wconstab - I guess I'm not too surprised that this test wobbled, since it seems like this test originally wobbled when we added symint clone'ing. Just wanted to see if you have any thoughts

Reply:
i'm trying to recall..

> [False, True, True, False, False]
this first block should correspond to the outputs of f, namely `return cat, sb, c, mm2`, where sb expands to 2 dims of size, hence True, True for is_sym_node

> + [False] * 4 + [True] * 4,
this part i can't remember exactly.  in the non-clone case we're saving one less non-sym-node for backward, and saving one more sym-node.  I think what i would do is check what these forward/backward graphs look like and what they were like before this PR.

Reply:
Hmm ok. The new graph does seem strictly better than the old one. To (partially) explain the difference of why we have -1 tensor and +1 symint saved:

- One thing I see is that we return `cat` twice in the old forward but not the new one.
- We also previously weren't saving one of the symints that was created in the forward, and now we are: `add: Sym(s1 + s2) = sym_size + sym_size_1`

Old:
```
class GraphModule(torch.nn.Module):
    def forward(self, primals_1: f32[s0], primals_2: f32[s1, s0], primals_3: f32[s2, s0], primals_4: f32[s0, 1]):
        # No stacktrace found for following nodes
        sym_size: Sym(s1) = torch.ops.aten.sym_size(primals_2, 0)
        sym_size_1: Sym(s2) = torch.ops.aten.sym_size(primals_3, 0)
        add: Sym(s1 + s2) = sym_size + sym_size_1
        sym_size_2: Sym(s0) = torch.ops.aten.sym_size(primals_1, 0)
        expand: f32[s1 + s2, s0] = torch.ops.aten.expand.default(primals_1, [add, sym_size_2]);  add = None
        cat: f32[2*s1 + 2*s2, s0] = torch.ops.aten.cat.default([expand, primals_2, primals_3])
        mm: f32[2*s1 + 2*s2, 1] = torch.ops.aten.mm.default(cat, primals_4)
        sym_size_3: Sym(1) = torch.ops.aten.sym_size(mm, 1)
        view: f32[1, s0] = torch.ops.aten.view.default(primals_1, [sym_size_3, sym_size_2]);  primals_1 = sym_size_3 = None
        mm_1: f32[2*s1 + 2*s2, s0] = torch.ops.aten.mm.default(mm, view)
        sym_size_5: Sym(s0) = torch.ops.aten.sym_size(primals_2, 1);  primals_2 = None
        return [cat, sym_size, sym_size_5, primals_3, mm_1, view, mm, primals_4, expand, cat, sym_size_2, sym_size, sym_size_1]
```

new:
```
class GraphModule(torch.nn.Module):
    def forward(self, primals_1: f32[s0], primals_2: f32[s1, s0], primals_3: f32[s2, s0], primals_4: f32[s0, 1]):
        # No stacktrace found for following nodes
        sym_size: Sym(s1) = torch.ops.aten.sym_size(primals_2, 0)
        sym_size_1: Sym(s2) = torch.ops.aten.sym_size(primals_3, 0)
        add: Sym(s1 + s2) = sym_size + sym_size_1
        sym_size_2: Sym(s0) = torch.ops.aten.sym_size(primals_1, 0)
        expand: f32[s1 + s2, s0] = torch.ops.aten.expand.default(primals_1, [add, sym_size_2])
        cat: f32[2*s1 + 2*s2, s0] = torch.ops.aten.cat.default([expand, primals_2, primals_3]);  expand = None
        mm: f32[2*s1 + 2*s2, 1] = torch.ops.aten.mm.default(cat, primals_4)
        sym_size_3: Sym(1) = torch.ops.aten.sym_size(primals_4, 1)
        view: f32[1, s0] = torch.ops.aten.view.default(primals_1, [sym_size_3, sym_size_2]);  primals_1 = sym_size_3 = None
        mm_1: f32[2*s1 + 2*s2, s0] = torch.ops.aten.mm.default(mm, view)
        sym_size_4: Sym(s0) = torch.ops.aten.sym_size(primals_2, 1);  primals_2 = None
        return [cat, sym_size, sym_size_4, primals_3, mm_1, view, mm, cat, primals_4, sym_size_1, sym_size, add, sym_size_2]
```

Reply:
hmm, aren't you returning cat twice in both old/new?